### ISSUE TITLE
tried it on a RG503 but no sound 

### ISSUE THREAD

**Issue by sostella** [jasonporritt/rg351_m8c#8]
Hi,
I have a RG503 and tried on it. I managed to install arkOS. I plugged the teensy with M8. M8 is working except from the sound. The sound looks ok in the mixer view. I tried the 3 commands you told about with and without sudo. it says deamon startup failed. written in red.
Would you have an idea of what I could do for this to work maybe?

**Comment by sostella** [comment]
Ok it seems that I did not have the teensy well plugged in when installing pulse.
Problem solved

**Comment by kolyan94** [comment]
I had the same issue with 353p, so maybe they have the same chip inside =(

**Comment by smootalicious** [comment]
This could be an issue with the order in which the audio is enumerated on the 503 and 353.  You could try a custom alsaloop command. See below.

Commands:
card# depends on order connected to the rg351
For best results, plug in teensy first, then external sound card then midi

`aplay -l` 
shows playback devices

`arecord -l`
shows capture devices

Example for aplay and arecord
	M8 is connected then external USB Audio device is connected

 	ark@rg351v:/roms2/ports/M8$ aplay -l

	**** List of PLAYBACK Hardware Devices ****
	card 0: rockchiprk817co [rockchip,rk817-codec], device 0: ff070000.i2s-rk817-hifi rk817-hifi-0 []
	  Subdevices: 0/1
	  Subdevice #0: subdevice #0
	card 1: M8 [M8], device 0: USB Audio [USB Audio]
	  Subdevices: 1/1
	  Subdevice #0: subdevice #0
	card 2: Device [USB Sound Device], device 0: USB Audio [USB Audio]
	  Subdevices: 1/1
	  Subdevice #0: subdevice #0

	ark@rg351v:/roms2/ports/M8$ arecord -l

	**** List of CAPTURE Hardware Devices ****
	card 0: rockchiprk817co [rockchip,rk817-codec], device 0: ff070000.i2s-rk817-hifi rk817-hifi-0 []
	  Subdevices: 1/1
	  Subdevice #0: subdevice #0
	card 1: M8 [M8], device 0: USB Audio [USB Audio]
	  Subdevices: 1/1
	  Subdevice #0: subdevice #0
	card 2: Device [USB Sound Device], device 0: USB Audio [USB Audio]
	  Subdevices: 1/1
	  Subdevice #0: subdevice #0

Connect M8 to rockchip out 
	`alsaloop -P hw:0,0 -C hw:1,0 -t 200000 -A 5 --rate 44100 --sync=0 -T -1 -d`

**Comment by smootalicious** [comment]
### **PulseAudio**

Determine active audio ports on USB audio
If you get errors, you may need to restart pulse before each command.

**Input ports**

First stop emulationstation
`ark@rg351v:/roms2/ports/M8$ sudo systemctl stop emulationstation`

```
ark@rg351v:/roms2/ports/M8$ sudo pulseaudio --start
W: [pulseaudio] main.c: This program is not intended to be run as root (unless --system is specified).


ark@rg351v:/roms2/ports/M8$ sudo pacmd
>>> list-sources
4 source(s) available.
...
      ports:
                analog-input-mic: Microphone (priority 8700, latency offset 0 usec, available: unknown)
                        properties:
                                device.icon_name = "audio-input-microphone"
                analog-input-linein: Line In (priority 8100, latency offset 0 usec, available: unknown)
                        properties:

        active port: <analog-input-linein>
...
```		
to change active input port:
`>>> set-source-port <index#> analog-input-linein`
or
`>>> set-source-port <index#> analog-input-mic`

**Output Ports**
`>>>list-sinks`
Note the sinks

Exit pacmd
`>>> exit`

Restart emulationstation
`ark@rg351v:/roms2/ports/M8$ sudo systemctl start emulationstation`

Edit the pulseaudio_pa file - set the sink and source corresponding to the correct device index from the listings in pacmd

**Comment by kolyan94** [comment]
Thanks, unfortunately, I sold my 353p 
Now using 351v and very happy.
I found that no matter what Linux-based OS I had in 353p when I turned on the console with the plugged-in headless I had no sound in the system at all.
Maybe for the future 503/353v/p/mp users, we could add some hints to the readme file.
For myself, I decided to save some money and buy a real m8 because now they are available every Monday =)
Just need to get the next salary, forgot about taxes (+150Є) =(
